feat(docker): Added support for known ECR tokens (WIP) (#20320)
renovatebot/renovate: docs/usage/docker.md

#### AWS ECR (Amazon Web Services Elastic Container Registry)

Renovate can authenticate with AWS ECR using AWS credentials, login password or an authorization token.

##### Using AWS account credentials

> To authenticate using AWS credentials, specify the access key id & secret as the username & password. If required
> define the session token also, for example:

Reviewer: ```suggestion
To authenticate using AWS credentials, you must use:

- the access key id as your username
- the secret as your pasword

If needed, you can set a session token, for example:
```

Please check if this suggestion is correct before applying it. The idea here is to put the terms next to their replacements, this should help users to use the right term in the right place.

How does a Renovate user know "If required define the session token also" applies to them? Do we need to explain this in the Renovate docs maybe?

Author: The suggestion looks good, thanks!

The token requirement depends on how the AWS credentials are issued...  It is possible to issue credentials both with and without requiring a session token.  It is typically managed by the AWS account admin.

}
```

##### Using the login password (basic auth)

> To authenticate using basic auth, simply provide the password in your config. For example:

Reviewer: ```suggestion
To authenticate with basic authentication, put the encrypted password in your config.
For example:
```

```json
{
  "hostRules": [
    {
      "hostType": "docker",
      "matchHost": "12345612312.dkr.ecr.us-east-1.amazonaws.com",
      "encrypted": {
        "password": "w...A"
      }
    }
  ]
}
```

> To obtain a login password, using the command `aws ecr get-login-password` or the corresponding API.

Reviewer: ```suggestion
To get a login password, use the command `aws ecr get-login-password` or the corresponding API.
```

Do we need to link to the API docs for this? Or do we expect AWS users to know this already?

Author: I would generally assume that anyone using Codeartifact would know about the AWS credentials for CLI or API use.  We can certainly link to the docs if it helps with clarity:
https://docs.aws.amazon.com/general/latest/gr/aws-sec-cred-types.html

##### Using an authorization token

> To authenticate using an authorization token, provide only the token in your config. For example:

Reviewer: ```suggestion
To authenticate with an authorization token, put the token in your config.
For example:
```

```json
{
  "hostRules": [
    {
      "hostType": "docker",
      "matchHost": "12345612312.dkr.ecr.us-east-1.amazonaws.com",
      "encrypted": {
        "token": "QV...09"
      }
    }
  ]
}
```

> To obtain an authorization token, using the command `aws ecr get-authorization-token` or the corresponding API.

Reviewer: ```suggestion
To get an authorization token, use the command `aws ecr get-authorization-token` or the corresponding API.
```

Same question here, do we expect AWS users to know about the API, or should we link to the docs?
